### DISCUSSION TITLE
Broken Script

### DISCUSSION THREAD

**Discussion by Alpha9624** [cc-tweaked/CC-Tweaked#1854] [Q&A]
Hi I came across this script and didnt't if anybody could fix it for the current version it used to be my main script I'd always use but now it's broken

Here's the pastebin if you wondering and thinking about fixing it: https://pastebin.com/6qmBBLpz

Thank you

**Comment by Lupus590** [comment]
Define broken.
What should it do that it's not doing? What does it do that it shouldn't be doing?
Do any error messages show? If yes then what is this error message?

**Comment by SquidDev** [comment]
The issue here is that the program hard codes several key codes. These [changed in Minecraft 1.13](https://tweaked.cc/reference/breaking_changes.html#mc-1.13), hence it no longer working on newer versions.

You can fix this by making the following changes:
 - On line 479, 545 and 626, replace `28` with `keys.enter`.
 - On line 481 replace `46` with `keys.c`
 - On line 489 replace `18` with `keys.e`
 - On line 498 replace `20` with `keys.t`
 - On line 506 replace `25` with `keys.p`.

Alternatively, you could use something like [this program](https://github.com/SquidDev-CC/random-programs/blob/master/key-code-converter.lua) to map the old keys to the new ones. So run `wget https://github.com/SquidDev-CC/random-programs/blob/master/key-code-converter.lua` to download, then run your mining script with `key-code-converter NAME_OF_THE_MINER_SCRIPT`.
